utils/cipher.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef CIPHER_H
+#define CIPHER_H
+
+#include <stdint.h>
+#include "common/defined_error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AES_BLOCK_SIZE 16
+#define AES_CBC_KEY_SIZE AES_BLOCK_SIZE * 2
+#define MAXLINE 1024
+#define IMSI_LEN 15
+
+/**
+ * @brief Encrypt plaintext
+ *
+ * @param[in] plaintext The text to be encrypted
+ * @param[in] plaintext_len Plaintext length
+ * @param[out] ciphertext Ciphrtext
+ * @param[in] ciphertext_len Ciphertext length
+ * @param[in, out] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[in] key[AES_CBC_KEY_SIZE] Encryption key
+ * @param[in] device_id[IMSI_LEN + 1] Device id
+ *
+ * @return
+ * - Ciphertext text new length on success
+ * - -1 on error
+ */
+int encrypt(const char *plaintext, int plaintext_len, char *ciphertext, int ciphertext_len, uint8_t iv[AES_BLOCK_SIZE],
```

Comment:
It is confusing to have both `encrypt` and `aes_encrypt` as public functions.

Reply:
Should we merge encrypt and aes_encrypt together?

Reply:
Always make public functions explicitly unique and usable.

Reply:
Not resolved yet.